### DISCUSSION TITLE
What are the advantages of setting proj_xy?

### DISCUSSION THREAD

**Discussion by vansin** [open-mmlab/mmrotate#725] [Q&A] [answered]
https://github.com/open-mmlab/mmrotate/blob/49228562a182fac21f5aa12b8a49e04f592e1f04/mmrotate/models/task_modules/coders/delta_xywht_rbbox_coder.py#L165-L170

oc define is False, and le90 and le135 define is True

![image](https://user-images.githubusercontent.com/25839884/218122338-0cbce274-5b06-4843-addc-53a702db22be.png)

**Comment by yangxue0827** [comment, marked as answer]
这个应该是有一段历史的：
oc定义法应该是我最早在做舰船检[1]的时候引入的，当时采用的框的编解码是最简单那种，也就是对应`proj_xy=False`；
le135和le90其实都是长边定义法只不过取值范围不同，le135最早我实在RRPN[2]文章中看到的，至于它们都用了`proj_xy=True`，是出自RoI Transformer[3]文章的设置：
<img width="825" alt="image" src="https://user-images.githubusercontent.com/29257168/218129872-13d7e73f-8594-4aa6-b678-c33fbab358b5.png">
<img width="813" alt="image" src="https://user-images.githubusercontent.com/29257168/218129971-ba0e8460-a93f-46e2-bb7c-cf13087ba451.png">
主要是考虑相对偏移量。

le135和le90之前其实还有一个差异，就是`edge_swap`，le135设置的是`False`，le90则是`True`。总结一下就是
oc：`edge_swap=False, proj_xy=False`
le90: `edge_swap=True, proj_xy=True`
le135: `edge_swap=False, proj_xy=True`
mmrotate都是按照原始出处设置的，其实所有oc的方法采用le90或le135的设置都会有性能的提升。

[1] Automatic ship detection in remote sensing images from google earth of complex scenes based on multiscale rotation dense feature pyramid networks, Remote Sensing, 2018
[2] Arbitrary-Oriented Scene Text Detection via Rotation Proposals, TMM, 2018
[3] Learning RoI Transformer for Detecting Oriented Objects in Aerial Images, CVPR, 2019
